0008-S-sidecar-endpoint-tls.md

```diff
@@ -0,0 +1,118 @@
+
+### Current Shortfalls
+- Inconsistency on setting up Dapr's sidecar endpoint on each SDK.
+- Not every SDK support a secure endpoint.
+
+## Related Items
+
+### Related proposals 
+
+Formalizing the proposal here from [this issue](https://github.com/dapr/dapr/issues/6035).
+
+## Expectations and alternatives
+
+* What is in scope for this proposal?
+- SDKs to support a consistent pair of environment variables to setup Dapr API
+- SDKs to support TLS endpoints for Dapr API
+
+* What is deliberately *not* in scope?
+- SSL certificate pinning
+- Have consistency of other environment variables for SDK (`DAPR_HOST`, `DAPR_SIDECAR_IP`, etc)
+- Have consistency of how Dapr client is instanciated on each SDK
+
+
+* What alternatives have been considered, and why do they not solve the problem?
+1. Leave every SDK as-is:
+  - Not every SDK offers an environment variable to configure Dapr endpoint, forcing configuration in code
+  - Environment variables per SDK, forcing sysadmin to know about each application's language use
+  - Not every SDK supports TLS endpoint
+2. Add TLS support only, giving each SDK room to decide on how to expose it to the user
+  - Not every SDK offers an environment variable to configure Dapr endpoint, forcing configuration in code
+  - Environment variables per SDK, forcing sysadmin to know about each application's language use
+
+* Are there any trade-offs being made? (space for time, for example)
+1. Leaving existing environment variables for host and port as-is per SDK, but driving consistency on this new way.
+2. Not changing Dapr's DAPR_HOST (or equivalent), DAPR_HTTP_PORT and DAPR_GRPC_PORT.
+
+* What advantages / disadvantages does this proposal have? 
+Pros:
+- Bring consistency in Dapr API endpoint configuration cross SDKs
+- Add support for TLS endpoint
+
+Cons:
+- Does not address existing inconsistencies in client instantiation and env variables
+- Needs to define a priority between new env variables and old ones
+
+## Implementation Details
+
+### Design
+
+* `DAPR_GRPC_ENDPOINT` defines entire endpoit for gRPC, not just host: `https://dapr-grpc.mycompany.com`
+* `DAPR_HTTP_ENDPOINT` defines entire endpoit for HTTP, not just host: `https://dapr-http.mycompany.com`
+* Port is parsed from the URL (`https://dapr.mycompany.com:8080`) or via the default port of the protocol used in the URL (80 for `http` and 443 for `https`)
+* `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT` can be set at the same time since some SDKs (Java, as of now) supports both protocols at the same time and app can pick which one to use.
+* `DAPR_GRPC_ENDPOINT` and `DAPR_HTTP_ENDPOINT` must be parsed and the protocol will be used for SDK to determine if communication is over TLS (if not done automatically). In summary, `https` means secure channel.
+* Initially, only `http` and `https` protocols should be supported. Future protocols can be added in the future depending on each language support.
```

Comment:
By Dapr, you mean runtime? The proposal does not affect runtime.

This proposal does not include support for H2C but does not close the door for it in the future.

Reply:
> By Dapr, you mean runtime? The proposal does not affect runtime.

Yes, the Dapr runtime doesn't support HTTPS. So if using `https` and protocol is HTTP, the connection will fail because the server doesn't have a HTTPS listener

Reply:
Any webserver including Dapr's can be fronted by ingress components (reverse proxies, load balancers etc) so while Dapr's server might not support it this enabled the former use cases.

Reply:
Exactly. This does not mean https directly to the sidecar.